index.js

```diff
@@ -37,24 +46,45 @@ const validate = (file, options) => {
 	return options;
 };
 
+const toArray = input => Array.isArray(input) ? input : [input];
+
 exports.buffer = async (file, options) => {
 	options = validate(file, options);
 
-	const isPid = typeof file === 'number';
+	const files = toArray(file);
+	const isPid = files.map(file => (typeof file === 'number').toString());
+
+	const {stdout} = await execFileP(bin, [files, options.size, isPid], spawnOptions);
 
-	const {stdout} = await execFileP(bin, [file, options.size, isPid], spawnOptions);
+	const buffers = stdout
+		.toString('hex')
+		.split(EOF_HEX)
+		.filter(s => Boolean(s))
+		.map(hex => Buffer.from(hex, 'hex'));
 
-	return stdout;
+	return buffers.length === 1 && !Array.isArray(file) ?
+		buffers[0] :
+		buffers;
 };
 
 exports.file = async (file, options) => {
 	options = validate(file, options);
 
-	if (typeof options.destination !== 'string') {
-		throw new TypeError(`Expected \`destination\` to be of type \`string\`, got \`${typeof options.destination}\``);
+	const isArray = Array.isArray(file);
+
+	if (typeof file === 'string' && typeof options.destination !== 'string') {
+		throw new TypeError(`Expected \`options.destination\` to be of type \`string\` when \`file\` is of type \`string\`, got \`${typeof options.destination}\``);
+	} else if (isArray && !Array.isArray(options.destination)) {
+		throw new TypeError(`Expected \`options.destination\` to be of type \`array\` when \`file\` is of type \`array\`, got \`${typeof options.destination}\``);
+	} else if (isArray && file.length !== options.destination.length) {
+		throw new TypeError('Expected `file` and `options.destination` arrays to be of the same length');
 	}
 
-	const isPid = typeof file === 'number';
+	const files = toArray(file);
+	const isPid = files.map(file => (typeof file === 'number').toString());
+	const destination = isArray ?
+		options.destination.join(',') :
```

Comment:
What if one of the destination paths has a `,`?

Reply:
I think the `execFileP` input is complex enough that it's best to just construct an object, `JSON.stringify()` it, and pass it as a single argument.

Reply:
> What if one of the destination paths has a ,?

It will fail

> I think the execFileP input is complex enough that it's best to just construct an object, JSON.stringify() it, and pass it as a single argument.

👍